packages/ckeditor5-build-decoupled-document/src/plugins/htmlInsert/htmlInsert.ts

```diff
@@ -0,0 +1,30 @@
+import { Plugin } from 'ckeditor5/src/core';
+import ButtonView from '@ckeditor/ckeditor5-ui/src/button/buttonview';
+import icon from './icon.svg';
+
+export default class HtmlInsert extends Plugin {
+	public init(): void {
+		const editor = this.editor;
+		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+		// @ts-ignore
+		const open = editor.config._config.htmlInsert.onOpen;
+		// The button must be registered among the UI components of the editor
+		// to be displayed in the toolbar.
+		editor.ui.componentFactory.add( 'htmlInsert', () => {
+			// The button will be an instance of ButtonView.
+			const button = new ButtonView();
+
+			button.set( {
+				label: 'HTML insert',
+				withText: false,
+				icon
+			} );
+
+			button.on( 'execute', () => {
+				open();
+			} );
+
+			return button;
+		} );
```

Comment:
Would add an if around it

```suggestion
		if ( editor.config._config.htmlInsert && editor.config._config.htmlInsert.onOpen ) {
			const open = editor.config._config.htmlInsert.onOpen;
			// The button must be registered among the UI components of the editor
			// to be displayed in the toolbar.
			editor.ui.componentFactory.add( 'htmlInsert', () => {
				// The button will be an instance of ButtonView.
				const button = new ButtonView();

				button.set( {
					label: 'HTML insert',
					withText: false,
					icon
				} );

				button.on( 'execute', () => {
					open();
				} );

				return button;
			} );
		}
```